src/WaveTrack.cpp

```diff
@@ -64,6 +64,37 @@ from the project that will own the track.
 
 using std::max;
 
+namespace {
+
+bool AreAligned(const WaveClipHolders& a, const WaveClipHolders& b)
+{
+    if (a.size() != b.size())
+        return false;
+
+    const auto less = [](const WaveClip* a, const WaveClip* b) {
+        return a->GetOffset() < b->GetOffset();
+    };
+
+    std::vector<WaveClip*> sorted;
+    sorted.reserve(a.size());
+    for (const auto& clip : b)
+    {
+        auto at = std::lower_bound(sorted.begin(), sorted.end(), clip.get(), less);
+        sorted.insert(at, clip.get());
+    }
```

Comment:
As mentioned, you might instead assume the two arrays of clips are already sorted.